overlord/snapstate/snapstate_test.go

```diff
@@ -5347,6 +5347,37 @@ func (s *snapmgrTestSuite) TestConflictExclusive(c *C) {
 	c.Check(err, ErrorMatches, `other changes in progress \(conflicting change "install-snap-a"\), change "create-recovery-system" not allowed until they are done`)
 }
 
+func (s *snapmgrTestSuite) TestOnlyConflictOnRelevantRunnableTask(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	chg := s.state.NewChange("some", "...")
+
+	// task that affects the same snap but won't run
+	failedTask := s.state.NewTask("failed", "")
+	failedTask.SetStatus(state.ErrorStatus)
```

Comment:
I think it would make sense to check against all tasks statuses in this test (you may want to use a clean state i.e `st := state.New()` on each iteration instead of s.state).